### DISCUSSION TITLE
Stopped functioning today?

### DISCUSSION THREAD

**Discussion by MrNickIE** [klejejs/ha-thermia-heat-pump-integration#102] [General]
Hello,

Thanks for this plugin.  I have been using for years.

Today, it stopped working, but appears to still be logged into our thermia online account.  This is soon after a HA upgrade to latest version.  Any ideas?

I have removed the working integration, that had a device with everything unavailable.  I have reinstalled the HACS addin again, and rebooted and now it is simply not detecting any device at all.

**Comment by MrNickIE** [comment]
Just as some additional information, the logs look ok?

2024-11-27 18:07:09.632 DEBUG (MainThread) [custom_components.thermia] Finished fetching thermia data in 0.001 seconds (success: True)

**Comment by ReconOperator** [comment]
There was some maintenance done today by Thermia so this might be the reason @klejejs ?

![Capture](https://github.com/user-attachments/assets/af6f185f-76e4-49bf-8fcf-0f2f63df1ad9)

**Comment by MrNickIE** [comment]
Oh my, I do pick my days :)

**Comment by flexi80** [comment]
I've had issues over the last few days where is seems like the integration freezes, or at least the operational status, a restart of the integration quickly fixes this.

**Comment by ReconOperator** [comment]
But i check my installation in HA and the logs.

2024-11-27 19:11:49.486 ERROR (SyncWorker_15) [ThermiaOnlineAPI.api.ThermiaAPI] Error getting device by id: 1528732
2024-11-27 19:12:00.435 ERROR (SyncWorker_25) [ThermiaOnlineAPI.api.ThermiaAPI] Error getting device by id: 1528732
2024-11-27 19:12:11.456 ERROR (SyncWorker_21) [ThermiaOnlineAPI.api.ThermiaAPI] Error getting device by id: 1528732
2024-11-27 19:12:22.469 ERROR (SyncWorker_47) [ThermiaOnlineAPI.api.ThermiaAPI] Error getting device by id: 1528732

I did a reload of the integration after i saw that and now it is unavailable

I had now issue before Thermia did something on their side

**Comment by MrNickIE** [comment]
The restart or full reboot isnt fixing it this time.  I am willing to bet it is the thermia side.

**Comment by klejejs** [comment]
Yeah, by the sound of it, it seems Thermia server-related, and, unfortunately there is nothing we can do about it 😞.

**Reply by ReconOperator** [reply]
Will you have time to look into what might have changed later on?

**Reply by klejejs** [reply]
I do not store or keep track of all the data and formats that Thermia uses; they are coded within the integration and the API which it uses.
I checked my heat pump and saw no issues with it, otherwise, I trust you, the users, will check your side, if something is not working, report it and then I can look into fixing the changes 😉.

**Reply by MrNickIE** [reply]
I am willing (hoping) that this issue will just go away :)

**Reply by klejejs** [reply]
@MrNickIE are you still experiencing the issue?

**Reply by MrNickIE** [reply]
Yes, nothing showing up, but the integration is running and does not appear to be getting any data out of my account.
2024-11-27 21:31:19.472 DEBUG (MainThread) [custom_components.thermia] Finished fetching thermia data in 0.001 seconds (success: True)

![image](https://github.com/user-attachments/assets/2836d47c-0175-4e39-8cbe-3d26255c6065)

**Reply by ReconOperator** [reply]
For me everything is unavailable 

@klejejs did you reload your integration and see what happens?

**Reply by klejejs** [reply]
No, I did not and, I had missed this.
I just released a new version `5.2.0` that should fix the issue. In case the issue still persists for you or you find another problem, please let me know.

**Reply by MrNickIE** [reply]
I wish I was a coder :)

Thanks for fixing whatever you fixed!  I will try to force an update now, though I am not sure how to do that :)

**Reply by MrNickIE** [reply]
FYI, all appears to be back working, will report back if I find anything odd!

**Reply by klejejs** [reply]
@MrNickIE There is an "Update information" option in top right menu of the integration in HACS, after you click that, HA should tell you there is a new version available.
![image](https://github.com/user-attachments/assets/d6c89f06-4c93-41b7-baea-7fe7e0108d73)

**Reply by ReconOperator** [reply]
Everything seems to work again. Big Thanks @klejejs 👍♥️

**Comment by robertmosen** [comment]
Can confirm that update is working.
